cmake/detect_features.cmake

```
@@ -28,7 +28,20 @@ macro(detect_cuda)
 endmacro(detect_cuda)
 
 macro(detect_openmp)
+
+    set( _find_openmp FALSE )
     if (NOT DEFINED GT_ENABLE_BACKEND_X86 AND NOT DEFINED GT_ENABLE_BACKEND_MC AND NOT DEFINED GT_ENABLE_BACKEND_NAIVE)
+        set( _find_openmp TRUE )
+    elseif (GT_ENABLE_BACKEND_X86 OR GT_ENABLE_BACKEND_MC OR GT_ENABLE_BACKEND_NAIVE)
+        set( _find_openmp TRUE )
+    endif()
+
+    option( GT_ENABLE_OPENMP "Enable OpenMP backends" ${_find_openmp} )
+
```

Let's make it an advanced option. What do you think @lukasm91 
```suggestion
    mark_as_advanced(GT_ENABLE_OPENMP)

```